### ISSUE TITLE
Problem with beacon diskusage on windows platform in 2018.3

### ISSUE THREAD

**Issue by skjaro** [saltstack/salt#47267]
### Description of Issue/Question
There is problem with beacon diskusage on windows machines with version 2018.3 of salt. Problem there is with escaping character on the end of mount point there is 'c:\\'. New version matches mount points with regex all the time. Older version 2017, matches mount point only when mount point does not exists. I don't know reason for that change. Solution is update documentation on windows which is now possible with:
```
beacons:
   diskusage:
      - interval: 60
      - 'C:\\': 90%
```
Or fix beacon to not check all the time with regex or escape it.

### Setup
diskusage.sls
```
beacons:
   diskusage:
      - interval: 60
      - 'C:\': 90%
```
### Steps to Reproduce Issue
```
Traceback (most recent call last):
  File "C:\salt\bin\lib\site-packages\salt\minion.py", line 2484, in handle_beacons
    beacons = self.process_beacons(self.functions)
  File "C:\salt\bin\lib\site-packages\salt\minion.py", line 444, in process_beacons
    return self.beacons.process(b_conf, self.opts['grains'])  # pylint: disable=no-member
  File "C:\salt\bin\lib\site-packages\salt\beacons\__init__.py", line 112, in process
    raw = self.beacons[fun_str](b_config[mod])
  File "C:\salt\bin\lib\site-packages\salt\beacons\diskusage.py", line 92, in beacon
    if re.match(mount, part.mountpoint):
  File "C:\salt\bin\lib\re.py", line 141, in match
    return _compile(pattern, flags).match(string)
  File "C:\salt\bin\lib\re.py", line 251, in _compile
    raise error, v # invalid expression
error: bogus escape (end of line)
```
### Versions Report
```
    Salt Version:
               Salt: 2018.3.0
     
    Dependency Versions:
               cffi: 1.10.0
           cherrypy: 10.2.1
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.3
          gitpython: Not Installed
              ioflo: Not Installed
             Jinja2: 2.10
            libgit2: Not Installed
            libnacl: 1.6.1
           M2Crypto: Not Installed
               Mako: 1.0.7
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.18
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:19:30) [MSC v.1500 32 bit (Intel)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.3
            timelib: 0.2.4
            Tornado: 4.5.2
                ZMQ: 4.1.6
     
    System Versions:
               dist:   
             locale: cp1250
            machine: x86
            release: 7
             system: Windows
            version: 7 6.1.7601 SP1 Multiprocessor Free
```

**Comment by gtmanfred** [comment]
'\' is an escape sequence in regexes, can you try setting it to 
```
beacons:
   diskusage:
      - interval: 60
      - 'C:\\': 90%
```

This could just be a documentation issue.

Daniel

**Comment by gtmanfred** [comment]
Ahh, nevermind, you already said that, would you mind opening a PR to correct the documentation?

Thanks,
Daniel

**Comment by gtmanfred** [comment]
It looks like it was included for the first time in this PR

https://github.com/saltstack/salt/pull/38935

Which was included in salt for the first time in 2017.7.5, that was when regex matching was first added.

So it would be new, just needs the documentation updated.

**Comment by skjaro** [comment]
It looks like change was in this PR #42317 and there is no mentioning about change only about new change from dictionary based config to list based config.

**Comment by skjaro** [comment]
I have fixed documentation in #47284. Maybe it is possible write regex without starting ^ when re.match is used in code because re.match already do this.
